internal/repositories/equipment.go

```diff
@@ -484,3 +485,90 @@ func OptionalIntsPetKind(k []int64, field string) predicate.PetKind {
 		s.Where(sql.InInts(s.C(field), petKind...))
 	}
 }
+
+func (r *equipmentRepository) BlockEquipment(
+	ctx context.Context, id int, startDate, endDate time.Time, userID int) error {
+	tx, err := middlewares.TxFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	eqToBlock, err := tx.Equipment.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	// Get EquipmentStatusName form DB
+	eqStatusNotAvailable, err := tx.EquipmentStatusName.
+		Query().
+		Where(equipmentstatusname.Name(domain.EquipmentStatusNotAvailable)).
+		Only(ctx)
+	if err != nil {
+		return err
+	}
+
+	start, end, err := checkDates(&startDate, &endDate)
+	if err != nil {
+		return err
+	}
+
+	// Set a new EquipmentStatusName for current Equipment
+	_, err = eqToBlock.Update().SetCurrentStatus(eqStatusNotAvailable).Save(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Create a new EquipmentStatus and set startDate, endDate, Equipment and EquipmentStatusName
+	_, err = tx.EquipmentStatus.Create().
+		SetCreatedAt(time.Now()).
+		SetEndDate(*end).
+		SetStartDate(*start).
+		SetEquipments(eqToBlock).
+		SetEquipmentStatusName(eqStatusNotAvailable).
+		SetUpdatedAt(time.Now()).
+		Save(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Get OrderStatusName form DB
+	orStatusBlocked, err := tx.OrderStatusName.
+		Query().
+		Where(orderstatusname.Status(domain.OrderStatusBlocked)).
+		Only(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Find all Orders which have OrderStatusName booked and start from startDate and later
+	orders, err := eqToBlock.QueryOrder().
+		Where(order.RentStartGTE(*start), order.RentStartLTE(*end)). // rentStart must be in range of startDate..endDate
+		Where(order.RentEndGTE(*start)).                             // rentEnd must be equal or greater that startDate
+		Where(order.HasOrderStatusWith(orderstatus.
+			HasOrderStatusNameWith(orderstatusname.StatusEQ(domain.OrderStatusPrepared)))).
+		Where(order.HasOrderStatusWith(orderstatus.
+			HasOrderStatusNameWith(orderstatusname.StatusEQ(domain.OrderStatusApproved)))).
+		All(ctx)
+
+	// Set a new OrderStatusName for these Orders and create new OrderStatus for each Order
+	for _, order := range orders {
+		tx.OrderStatus.Create().
```

Comment:
Did you look at [CreateMany](https://entgo.io/docs/crud/#create-many) Ent's feature? Does it have any sense to use it here?

Reply:
I have no idea how to use CreateBulk or MapCreateBuk in this case.

Reply:
For example:
```
oss := make([]*ent.OrderStatusCreate, len(orderIDs))
	for i, order := range orderIDs {
		oss[i] = tx.OrderStatus.Create().
			SetCurrentDate(time.Now()).
			SetOrderID(order).
			SetUsersID(userID).
			SetComment("Eqipment blocked").
			SetOrderStatusName(orStatusBlocked)
	}
	_, err = tx.OrderStatus.CreateBulk(oss...).Save(ctx)
	if err != nil {
		return err
	}
```

This code generates the following SQL:
```
LOG:  execute lrupsc_1_13: INSERT INTO "order_status" ("comment", "current_date", "order_order_status", "order_status_name_order_status", "user_order_status") VALUES ($1, $2, $3, $4, $5), ($6, $7, $8, $9, $10) RETURNING "id"
DETAIL:  parameters: $1 = 'Eqipment blocked', $2 = '2023-08-29 13:16:19.184107', $3 = '3', $4 = '8', $5 = '8', $6 = 'Eqipment blocked', $7 = '2023-08-29 13:16:19.184109', $8 = '4', $9 = '8', $10 = '8'
```